cmd/honeyelb2/main.go

```diff
@@ -0,0 +1,249 @@
+          os.Exit(1)
+        }
+
+        lbArn := lbNameResp.LoadBalancers[0].LoadBalancerArn
+        lbArnResp, err := elbSvc.DescribeLoadBalancerAttributes(&elbv2.DescribeLoadBalancerAttributesInput{
+          LoadBalancerArn: lbArn,
+        })
+        if err != nil {
+          fmt.Fprintln(os.Stderr, err)
+          os.Exit(1)
+        }
+
+        enabled := false
+        bucketName := ""
+        bucketPrefix := ""
+
+        for _, element := range lbArnResp.Attributes {
+          if *element.Key == "access_logs.s3.enabled" && *element.Value == "true" {
+            enabled = true
+          }
+          if *element.Key == "access_logs.s3.bucket" {
+            bucketName = *element.Value
+          }
+          if *element.Key == "access_logs.s3.prefix" {
+            bucketPrefix = *element.Value
+          }
+        }
+
+        if !enabled {
+          fmt.Fprintf(os.Stderr, `Access logs are not configured for ALB %q. Please enable them to use the ingest tool.
+
+For reference see this link:
+
+http://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-access-logs.html#enable-access-logging
+`, lbName)
+          os.Exit(1)
+        }
+        logrus.WithFields(logrus.Fields{
+          "bucket": bucketName,
+          "lbName": lbName,
+        }).Info("Access logs are enabled for ALB ♥")
+
+        elbDownloader := logbucket.NewELBDownloader(sess, bucketName, bucketPrefix, lbName, "alb")
+        downloader := logbucket.NewDownloader(sess, stater, elbDownloader, opt.BackfillHr)
+
+        // TODO: One-goroutine-per-LB feels a bit
+        // silly.
+        go downloader.Download(downloadsCh)
+      }
+
+      signalCh := make(chan os.Signal)
+      signal.Notify(signalCh, os.Interrupt)
+
+      go func() {
+        <-signalCh
+        logrus.Fatal("Exiting due to interrupt.")
+        // TODO(nathanleclaire): Cleanup before
+        // exiting.
+        //
+        // 1. Delete format file, even
+        //    though it's in /tmp.
+        // 2. Also, wait for existing in-flight object
+        //    parsing / sending to finish so that state of
+        //    parsing "cursor" can be written to the JSON
+        //    file.
+      }()
+
+      for {
+        download := <-downloadsCh
+        if err := defaultPublisher.Publish(download); err != nil {
+          logrus.WithField("object", download).Error("Cannot properly publish downloaded object")
+        }
+      }
+    }
+  }
+
+  return fmt.Errorf("Subcommand %q not recognized", args[0])
+}
+
+func main() {
+  flagParser := flag.NewParser(opt, flag.Default)
+  args, err := flagParser.Parse()
+  if err != nil {
+    os.Exit(1)
+  }
+
+  if opt.Debug {
+    logrus.SetLevel(logrus.DebugLevel)
+  }
+
+  formatter := &logrus.TextFormatter{
+    FullTimestamp: true,
+  }
+  logrus.SetFormatter(formatter)
+
+  logrus.WithField("version", BuildID).Debug("Program starting")
+
+  if opt.Dataset == "aws-$SERVICE-access" {
+    opt.Dataset = "aws-elb-access"
```

Comment:
Do we want logs from elb and alb going to the same dataset?

Reply:
My opinion - the difference between ELBs (eg the api vs. www ELB) is actually larger than the difference between an ELB and an ALB, so given that we send all ELBs to the same dataset, I think it's just fine to send ALB traffic into the same dataset.  
Might have a different response for NLBs since that traffic is pretty different...
As always, we could make it configurable, but I think the current "all to the same place" is fine, and if we want to make it configurable, can do so in a followup PR.